xdsl/interactive/app.py

```diff
@@ -347,10 +347,12 @@ def expand_node(
         # remove potential children nodes in case expand node has been clicked multiple times on the same node
         expanded_pass.remove_children()
 
-        for pass_name, value, value_spec in child_pass_list:
+        for pass_name, value in child_pass_list:
             expanded_pass.add(
                 label=pass_name,
-                data=(value, value_spec),
+                data=(type(value), value.pipeline_pass_spec())
+                if isinstance(value, ModulePass)
+                else (value, None),
```

Comment:
This is a newly introduced ugly bit due to the tree node data still being in terms of pass type and spec, next PR will remove this.